src/avram/charms/json_extensions.cr

```diff
@@ -78,17 +78,27 @@ struct JSON::Any
     class Criteria(T, V) < Avram::Criteria(T, V)
       # performs `WHERE jsonb ? string`
       def has_key(value : String) : T
-        add_clause(Avram::Where::JSONHasKey.new(column, value))
+        add_clause(Avram::Where::JsonbHasKey.new(column, value))
       end
 
       # performs `WHERE jsonb ?| array`
       def has_any_keys(keys : Array(String)) : T
-        add_clause(Avram::Where::JSONHasAnyKeys.new(column, keys))
+        add_clause(Avram::Where::JsonbHasAnyKeys.new(column, keys))
       end
 
       # performs `WHERE jsonb ?& array`
       def has_all_keys(keys : Array(String)) : T
-        add_clause(Avram::Where::JSONHasAllKeys.new(column, keys))
+        add_clause(Avram::Where::JsonbHasAllKeys.new(column, keys))
+      end
+
+      # performs `WHERE jsonb @> other_json`
+      def includes(other_json : JSON::Any)
+        add_clause(Avram::Where::JsonbIncludes.new(column, other_json.to_json))
+      end
+
+      # performs `WHERE jsonb <@ other_json`
```

Comment:
Do you think it's possible to add an overload of this method that allows passing in some less verbose type? JSON::Any is so incredibly verbose.

- `def in(other_json : NamedTuple(String, String))` 
- `def in(other_json : Hash(String, String))` 
- maybe even `def in(other_json : T)` -- and do some logic to stringify keys and values?

Reply:
Totally agree that `JSON::Any` is very verbose. It's also very difficult to construct in many cases without taking a performance hitting shortcut. However, postgres needs the value to look like JSON, so the structure just has to be able to fit in to that. I guess what I could do is leave it untyped, then just call `to_json`. Though, this just means it won't catch some errors at compile-time.

I would say just do `Hash` or `NamedTuple` explicitly, but if you have nested objects that need to be checked, then that wouldn't work. For example

```
# {"user": {"username": "lucky", "preferences": {"theme": "dark"}}}
# This would work with my current change
UserQuery.new.preferences.in(params.from_json)
```

But if we changed this to say `Hash(String, String)` or `NamedTuple`, then the above wouldn't work.

Reply:
It would be nice to be able to specify a duck-type here, wouldn't it?

Reply:
Any suggestions on what that might look like? It would have to be like `in(value : HasToJsonMethod)` :joy:  though... I think `Object` has `to_json` :thinking:  but that may be too broad. The other thing is that your `jsonb` field may be a json array like `[{"updated": "..."}, {"updated": "..."}]`, so this value would need to be any number of things like JSON::Any, Hash, Array(Hash), NamedTuple, Array(NamedTuple). Can we do `in(value : TellDevinToJustAssumeWhatImThinking)` yet? :rofl:

Reply:
The latter, yes. 😆